ProjFS.Mac/PrjFSLib/prjfs-log/kext-perf-tracing.cpp

```diff
@@ -51,20 +54,34 @@ bool PrjFSLog_FetchAndPrintKextProfilingData(io_connect_t connection)
     }
     else if (ret == kIOReturnSuccess)
     {
-        for (unsigned i = 0; i < Probe_Count; ++i)
+        printf("   Counter                             [ Samples  ][Total time (ns)][Mean (ns)   ][Stddev (ns) ][Min (ns)][Max (ns)  ]\n");
+        printf("----------------------------------------------------------------------------------------------------------------------\n");
+        
+        for (unsigned i = 0; i < PrjFSPerfCounter_Count; ++i)
         {
-            double samples = probes[i].numSamples1;
-            double sum_abs = probes[i].sum;
-            double stddev_abs = samples > 1 ? sqrt((samples * probes[i].sumSquares - sum_abs * sum_abs) / (samples * (samples - 1))) : 0.0;
-
-            double sum_ns = nanosecondsFromAbsoluteTime(sum_abs);
-            double stddev_ns = nanosecondsFromAbsoluteTime(stddev_abs);
-            double mean_ns = samples > 0 ? sum_ns / samples : 0;
-            printf("%2u %40s  %8llu [%8llu] samples, total time: %15.0f ns, mean: %10.2f ns +/- %11.2f",
-                i, PerfCounterNames[i], probes[i].numSamples1, probes[i].numSamples2, sum_ns, mean_ns, stddev_ns);
+            double samples = probes[i].numSamples;
+            printf(
+                "%2u %-35s [%10llu]",
+                i,
+                PerfCounterNames[i],
+                probes[i].numSamples);
+            
             if (probes[i].min != UINT64_MAX)
             {
-                printf(", min: %7llu ns, max: %10llu ns\n",  nanosecondsFromAbsoluteTime(probes[i].min), nanosecondsFromAbsoluteTime(probes[i].max));
+                double sum_abs = probes[i].sum;
+                double stddev_abs = samples > 1 ? sqrt((samples * probes[i].sumSquares - sum_abs * sum_abs) / (samples * (samples - 1))) : 0.0;
+
+                double sum_ns = nanosecondsFromAbsoluteTime(sum_abs);
+                double stddev_ns = nanosecondsFromAbsoluteTime(stddev_abs);
+                double mean_ns = samples > 0 ? sum_ns / samples : 0;
+
+                printf(
+                    "[%15.0f][%12.0f][%12.0f][%8llu][%10llu]\n",
+                    sum_ns,
```

Comment:
Should sum always be in ms rather than ns?

Reply:
Yes these result in large numbers, but I personally prefer keeping all of the units the same to make it easier to scan through the table and not have to do mental conversions to compare various numbers. If you prefer, I can convert everything to ms or microseconds. The risk is that some of the lower numbers round down to 0, but then again those are not the interesting ones when searching for hot spots.